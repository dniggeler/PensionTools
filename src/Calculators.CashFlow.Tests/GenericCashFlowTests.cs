using System.Collections.Generic;
using System.Linq;
using Calculators.CashFlow.Models;
using Snapshooter.Xunit;
using Xunit;

namespace Calculators.CashFlow.Tests
{
    [Trait("Higher Level Calculators", "Cash-Flow")]
    public class GenericCashFlowTests
    {
        [Fact(DisplayName = "Single Allocation for 10 Years")]
        public void Generate_Single_Allocation_For_10_Years()
        {
            // given
            GenericCashFlowDefinition definition = new GenericCashFlowDefinition
            {
                NetGrowthRate = 0,
                Name = "Test",
                InitialAmount = 10_000,
                RecurringAmount = (500, FrequencyType.Yearly),
                Flow = (FundsType.Exogenous, FundsType.TaxableWealth),
                InvestmentPeriod = (2021, 10)
            };

            CashFlowGenerator generator = new CashFlowGenerator();

            // when
            List<CashFlowModel> result = generator.Generate(definition).ToList();

            // then
            Snapshot.Match(result);
        }
    }
}
