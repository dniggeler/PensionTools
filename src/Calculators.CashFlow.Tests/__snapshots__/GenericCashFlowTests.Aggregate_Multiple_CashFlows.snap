[
  {
    "Year": 2021,
    "Amount": 30000.0,
    "Source": "Exogenous",
    "Target": "Wealth",
    "IsTaxable": true,
    "TaxType": "Wealth",
    "OccurrenceType": "BeginOfPeriod"
  },
  {
    "Year": 2022,
    "Amount": 1000.0,
    "Source": "Exogenous",
    "Target": "Wealth",
    "IsTaxable": true,
    "TaxType": "Wealth",
    "OccurrenceType": "BeginOfPeriod"
  },
  {
    "Year": 2023,
    "Amount": 1000.0,
    "Source": "Exogenous",
    "Target": "Wealth",
    "IsTaxable": true,
    "TaxType": "Wealth",
    "OccurrenceType": "BeginOfPeriod"
  },
  {
    "Year": 2024,
    "Amount": 1000.0,
    "Source": "Exogenous",
    "Target": "Wealth",
    "IsTaxable": true,
    "TaxType": "Wealth",
    "OccurrenceType": "BeginOfPeriod"
  },
  {
    "Year": 2025,
    "Amount": 1000.0,
    "Source": "Exogenous",
    "Target": "Wealth",
    "IsTaxable": true,
    "TaxType": "Wealth",
    "OccurrenceType": "BeginOfPeriod"
  },
  {
    "Year": 2026,
    "Amount": 500.0,
    "Source": "Exogenous",
    "Target": "Wealth",
    "IsTaxable": true,
    "TaxType": "Wealth",
    "OccurrenceType": "BeginOfPeriod"
  },
  {
    "Year": 2027,
    "Amount": 500.0,
    "Source": "Exogenous",
    "Target": "Wealth",
    "IsTaxable": true,
    "TaxType": "Wealth",
    "OccurrenceType": "BeginOfPeriod"
  },
  {
    "Year": 2028,
    "Amount": 500.0,
    "Source": "Exogenous",
    "Target": "Wealth",
    "IsTaxable": true,
    "TaxType": "Wealth",
    "OccurrenceType": "BeginOfPeriod"
  },
  {
    "Year": 2029,
    "Amount": 500.0,
    "Source": "Exogenous",
    "Target": "Wealth",
    "IsTaxable": true,
    "TaxType": "Wealth",
    "OccurrenceType": "BeginOfPeriod"
  },
  {
    "Year": 2030,
    "Amount": 500.0,
    "Source": "Exogenous",
    "Target": "Wealth",
    "IsTaxable": true,
    "TaxType": "Wealth",
    "OccurrenceType": "BeginOfPeriod"
  },
  {
    "Year": 2021,
    "Amount": 50000.0,
    "Source": "Exogenous",
    "Target": "CapitalBenefits",
    "IsTaxable": false,
    "TaxType": "Undefined",
    "OccurrenceType": "BeginOfPeriod"
  },
  {
    "Year": 2021,
    "Amount": 0.0,
    "Source": "Income",
    "Target": "CapitalBenefits",
    "IsTaxable": false,
    "TaxType": "Undefined",
    "OccurrenceType": "BeginOfPeriod"
  }
]
